sound/soc/sof/intel/hda.c

```diff
@@ -395,6 +400,11 @@ int hda_dsp_pre_fw_run(struct snd_sof_dev *sdev)
 					    0644, sdev->debugfs_root,
 					    &hdev->skip_imr_boot);
 		}
+
+		if (hda_keep_fw_dma_buffer == 0 || hda_keep_fw_dma_buffer == 1)
+			hdev->keep_fw_dma_buffer = hda_keep_fw_dma_buffer;
+		else
+			hdev->keep_fw_dma_buffer = !hdev->imrboot_supported;
```

Comment:
this is become hard to track, IIRC there was a case where if the firmware crashes in the previous run then IMR is disabled and we would need a DMA buffer.

Why not just allocate the DMA buffer always, even if IMR is used after the first boot.

Reply:
You mean keeping the DMA buffer unconditionally?

This does not change how we do IMR, it is only aimed to give an option that we can do the DMA buffer allocation and setup only once and if needed use it.

We can make this opt-out, so that by default we would keep the buffer and users can opt out?